src/pylibsshext/session.pyx

```diff
@@ -383,6 +393,19 @@ cdef class Session(object):
         if rc == libssh.SSH_AUTH_ERROR or rc == libssh.SSH_AUTH_DENIED:
             raise LibsshSessionException("Failed to authenticate with password: %s" % self._get_session_error_str())
 
+    def authenticate_interactive(self, password):
+        cdef int rc
+        rc = libssh.ssh_userauth_kbdint(self._libssh_session, NULL, NULL)
+        if rc == libssh.SSH_AUTH_INFO:
+            prompt_count = libssh.ssh_userauth_kbdint_getnprompts(self._libssh_session)
+            if prompt_count == 1:
+                rc = libssh.ssh_userauth_kbdint_setanswer(self._libssh_session, 1, password.encode())
+            else:
+                raise LibsshSessionException("Server aksed more questions than I was expecting (%s), bailing..." % prompt_count)
```

Comment:
Let's turn this into a guard expression and dedent some bits. Also, let's use str.format for forward compat:
```suggestion
            if prompt_count != 1:
                raise LibsshSessionException(
                    "Server aksed more questions than I was expecting "
                    "({prompt_num!s}), bailing...".
                    format(prompt_num=prompt_count)
                )

            rc = libssh.ssh_userauth_kbdint_setanswer(self._libssh_session, 1, password.encode())
```